test/e2e/e2e_test.go

```diff
@@ -292,6 +299,134 @@ func pollForConfigMapChange(client *kubernetes.Clientset, compareConfigMap *v1.C
+			Name:      s.Name,
+			Namespace: s.Namespace,
+			Labels:    s.Labels,
+		},
+		Spec: routev1.RouteSpec{
+			TLS: &routev1.TLSConfig{
+				InsecureEdgeTerminationPolicy: routev1.InsecureEdgeTerminationPolicyRedirect,
+				Termination:                   routev1.TLSTerminationPassthrough,
+			},
+			To: routev1.RouteTargetReference{
+				Kind: "Service",
+				Name: s.Name,
+			},
+			Port: &routev1.RoutePort{
+				TargetPort: s.Spec.Ports[0].TargetPort,
+			},
+		},
+	})
+
+	if err = wait.PollImmediate(5*time.Second, 5*time.Minute, func() (done bool, err error) {
+		localRoute, err := routeClient.Routes(namespace).Get(s.ObjectMeta.Name, metav1.GetOptions{})
+		if err != nil && errors.IsNotFound(err) {
+			return false, nil
+		}
+		if !IsRouteAdmitted(localRoute) {
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		return nil, err
+	}
+	return route, nil
+}
+
+func IsRouteAdmitted(route *routev1.Route) bool {
+	for _, ingress := range route.Status.Ingress {
+		for _, condition := range ingress.Conditions {
+			if condition.Type == routev1.RouteAdmitted && condition.Status == v1.ConditionTrue {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// createService creates a service object and returns a pointer pointing to this object on success.
+func createService(client *kubernetes.Clientset, namespace string, serviceName string, deploymentName string) (*v1.Service, error) {
+	service, err := client.CoreV1().Services(namespace).Create(&v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      serviceName,
+			Namespace: namespace,
+			Labels:    map[string]string{"app": deploymentName},
+		},
+		Spec: v1.ServiceSpec{
+			Ports: []v1.ServicePort{
+				{
+					Port:     8443,
+					Protocol: v1.ProtocolTCP,
+					TargetPort: intstr.IntOrString{
+						Type:   intstr.Int,
+						IntVal: 8443,
+					},
+					Name: "https",
+				},
+			},
+			Type:     v1.ServiceTypeClusterIP,
+			Selector: map[string]string{"app": deploymentName},
+		},
+	})
+
+	if err != nil {
+		return nil, err
+	}
+	return service, nil
+}
+
+func testMetricsEndpoint(t *testing.T, client *kubernetes.Clientset, routeClient *routeclient.RouteV1Client, deploymentName string, serviceName string, namespace string) {
+	var res *http.Response
+	defer func() {
+		if res != nil {
+			res.Body.Close()
+		}
+		client.CoreV1().Services(namespace).Delete(serviceName, &metav1.DeleteOptions{})
+		routeClient.Routes(namespace).Delete(serviceName, &metav1.DeleteOptions{})
+	}()
+	service, err := createService(client, namespace, serviceName, deploymentName)
+	if err != nil {
+		t.Fatalf("could not create a service: %v", err)
+	}
+	route, err := createRouteForService(routeClient, service, namespace)
+	if err != nil {
+		t.Fatalf("could not create a route: %v", err)
+	}
+
+	url := fmt.Sprintf("https://%s/metrics", route.Spec.Host)
+
+	httpclient := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
```

Comment:
- This needs to be false. 
- A metrics endpoint needs to be secured with a service ca-provided serving cert. 
  - The metrics endpoint is automatically secured by a serving cert provided the pod template has the serving cert secret mounted to `/var/run/secrets/serving-cert` (this is already true for the operator).
  - The controller framework will generate an insecure self-signed certificate on startup and will exit the pod when the cert files are updated, so even the service ca controllers and operators can be secured with serving certs.  
  - Configuring each controller with a service ca-provided ca bundle will require creating an annotated service ( `service.beta.openshift.io/serving-cert-secret-name`) that exposes the metrics endpoint.
- The client to access the metrics endpoint needs to be configured with a service ca-provided ca bundle (see the rotation tests for how to accomplish this).

@stlaz Who is going to watch these new endpoints, and how do they discover them?

Reply:
Those endpoints are not new, they are being scraped by monitoring operator, they most probably use their service account for that.

The InsecureSkipVerify is here as you're not trying to see whether the endpoint is secured by a valid certificate, that it should always be, otherwise scraping the data would not work, it merely tested that you can't scrape it unauthenticated.

Reply:
The use of `InsecureSkipVerify` is masking a bug.  Metrics endpoints are not secured automatically by a valid certificate, as per my previous comment, and anyone trying to connect to the endpoint enabled by the current state of this PR is going to be unable to validate the cert.

Can you please point me to how the monitoring operator knows how to read from these endpoints? Given that the endpoint cert is not secure in the current iteration of this PR, I am interested to know how to verify on the monitoring operator side whether the endpoints it is charged with collecting from are configured properly.